### PR TITLE
ENG-165636 - Refactored into a Monorepo

### PR DESCRIPTION
**Pull request by mitcsutt**
This is a large change to the structure of the project, but not to the output.
- Created a monorepo
  - `packages/mds` is now the home of `@moxiworks/mds`
    - Stencil components
    - Tailwind styles
    - Docs 
  - `packages/mds-react` is now the home of `@moxiworks/mds-react`
    - A separate package to be published, but contains the output from `@moxiworks/mds`
- Stencil js now has multiple configs
  - `stencil.config.ts` - Contains all builds required
  - `stencil.config.docs.ts` - Builds only the output for the vuepress docs
  - `stencil.config.mds.ts` -Builds only the output for the `@moxiworks/mds` package output
  - `stencil.config.react.ts` - Builds only the output for the react into the react package
- All build outputs are now in the gitignore.
- pipeline.yml file created to add a CI pipeline
  - A docker image is built to be used for tests and test builds
  - ts check runs to make sure there are no ts issues
  - stencil test is run to test stencil components
  - We build just the mds output to make sure it doesn't fail
  - We build just the docs output to make sure it doesn't fail
  - We build just the react output to make sure it doesn't fail 
- On `main` we use a docker image that is prebuilt
  - We publish the docs directory to a `gh-pages` branch to be used by github pages

On top of this I had to fix a few issues such as ts problems and stencil tests that were already failing before this change

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,35 +1,6 @@
...
-*.sublime-project
-*.sublime-workspace
-
-.stencil/
-.nova/
-.idea/
-.vscode/
-.sass-cache/
-.versions/
+.vscode/*
```
Comment (mitcsutt) [inline .gitignore:1]: I'm letting each package define their own gitignore

File: .vscode/launch.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+      {
+        "type": "node-terminal",
+        "request": "launch",
+        "name": "Spec Test Current File",
+        "cwd": "${workspaceFolder}/packages/mds",
+        "command": "yarn test ${fileBasename}",
+        "internalConsoleOptions": "neverOpen",
+      },
+    ]
+}
```
Comment (mitcsutt) [inline .vscode/launch.json:21]: This allows you to debug stencil tests with vscodes inbuilt debugger.

File: README.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# Mercury Design System
```
Comment (mitcsutt) [inline README.md:1]: I will improve the documentation in later changes

File: ops/ci/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+ARG GITHUB_USERNAME
+ARG GITHUB_EMAIL
+ARG GITHUB_TOKEN
+
+RUN apk update && apk upgrade && \
+    apk add --no-cache bash git openssh
+
+RUN git config --global user.name ${GITHUB_USERNAME}
+RUN git config --global user.email ${GITHUB_EMAIL}
+RUN git remote set-url origin https://git:${GITHUB_TOKEN}@github.com/moxiworks/mds.git
```
Comment (mitcsutt) [inline ops/ci/Dockerfile:26]: Sets all the git details to allow publishing directly to the `gh-pages` branch fro the docs output

File: packages/mds-react/package.json [hunk 5]
```diff
@@ -0,0 +1,26 @@
+{
+  "name": "@moxiworks/mds-react",
+  "version": "2.0.0",
```
Comment (mitcsutt) [inline packages/mds-react/package.json:3]: I'm thinking of keeping them the same version as each other. Thoughts?

File: packages/mds/src/utils/transitions.ts [hunk 6]
```diff
@@ -157,7 +157,7 @@ function setStyleProperty(el: HTMLElement, property, value) {
     // Instead, we need to remove any conflicting transform values and then add the new value.
     // First, parse out an array of transforms (e.g. ['translate(532px, 311px)', 'scale(0.8)'])
     const matchTransforms = /\w*\((-?((\d+)|(\d*\.\d+))\w*,\s*)*(-?(\d+)|(\d*\.\d+))\w*\)/gi;
-    let transforms = el.style.transform.match(matchTransforms) || [];
+    let transforms = el.style.transform.match(matchTransforms) || [] as string[];
```
Comment (mitcsutt) [inline packages/mds/src/utils/transitions.ts:160]: This was a ts issue I had to fix

**Review by kevinleedrum — APPROVED** [review]
This is fantastic!  I love the use of yarn workspaces here.  👏👏👏

**Review by johnmcclumpha — APPROVED** [review]
Great work @mitcsutt 🎉

**Review by jesseduffield — APPROVED** [review]
LGTM, great work.

Worth considering if apdev would fit well here to spare users from having to install the correct npm/yarn/nvm versions. If devs working on mds are likely to be working on other moxi repos then I reckon it's a good fit, otherwise probably not

**Comment by jesseduffield** [comment]
Yeah for sure, I didn't mean to say it should happen now

**Pull request merged**